### PR TITLE
Feat/us15

### PR DESCRIPTION
**Pull request by FeliKroder**
What's changed:
- implement SearchBox using Fuse.js
- implement CSS Animation for SearchBox-Input
- change position of Filter-Button and SearchBox in ExpenseList

### CODE REVIEW

File: components/SearchBox/index.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+} from "@/design-system/StyledSearchBox";
+
+const fuseOptions = {
+  shouldSort: true,
+  minMatchCharLength: 1,
+  threshold: 0.3,
+  keys: ["description", "name", "amount"],
+};
+
+function SearchBox() {
```
Comment (Roland-Hufnagel) [inline components/SearchBox/index.jsx:19]: why not passing ``data`` as a prop instead of fetching it again?

File: components/SearchBox/index.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        onFocus={() => setResultsVisible(true)}
+        onBlur={() => setResultsVisible(false)}
+        autoComplete="off"
+      />
+
+      <StyledSearchIcon icon="material-symbols:search" width="24px" />
+      <StyledSearchOutput className={resultsVisible ? "show" : ""}>
+        {results.map((result) => (
+          <StyledLink href={`/expense/${result._id}`} key={result._id}>
+            <li>{result.name}</li>
```
Comment (Roland-Hufnagel) [inline components/SearchBox/index.jsx:68]: You should have the Link inside the ListItem.

File: components/ExpenseList/index.jsx [hunk 3]
```diff
@@ -11,6 +11,8 @@ import { useEffect, useState } from "react";
 import FilterExpense from "../FilterExpense";
 import ListItemPagination from "../ListItemPagination";
 import Expenses from "../Expenses";
+import HomePage from "../SearchBox";
```
Comment (Roland-Hufnagel) [inline components/ExpenseList/index.jsx:14]:
```suggestion

```
I think it was a mistake?

**Comment by BirkBee** [comment]
LGTM, a very nice app! 🙌 
looks very fancy and professional, keep up the good work 🚀

**Pull request merged**